src/pages/Swap/index.tsx

```diff
@@ -194,16 +225,34 @@ export default function Swap({ className }: { className?: string }) {
   // for expert mode
   const [isExpertMode] = useExpertModeManager()
   // swap state
-  const [state, dispatch] = useReducer(swapReducer, initialSwapState)
+  const [state, dispatch] = useReducer(swapReducer, { ...initialSwapState, ...prefilledState })
   const { typedValue, recipient, independentField } = state
+
+  const previousConnectedChainId = usePrevious(connectedChainId)
+  useEffect(() => {
+    const combinedInitialState = { ...initialSwapState, ...prefilledState }
+    console.log({ combinedInitialState })
+    if (previousConnectedChainId && previousConnectedChainId !== connectedChainId) {
+      dispatch(
+        replaceSwapState({
+          ...initialSwapState,
+          ...prefilledState,
+          field: combinedInitialState.independentField ?? Field.INPUT,
```

Comment:
is this needed? doesnt the spread above do the same? and isn't `independentField` always defined

Reply:
this is needed because the reducer action uses a different name (`field`) than the state type def (`independentField`). i wanted to avoid changing it to keep my changes minimal